realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SchemaTests.kt

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.realm.SyncTestUtils.Companion.createTestUser
+import io.realm.entities.StringOnly
+import junit.framework.Assert
+import junit.framework.TestCase
+import org.junit.*
+import org.junit.runner.RunWith
+import kotlin.test.assertFailsWith
+import kotlin.test.assertTrue
+
+@RunWith(AndroidJUnit4::class)
+class SchemaTests {
+    @get:Rule
+    val configFactory = TestSyncConfigurationFactory()
+
+    private var config: SyncConfiguration? = null
```

Comment:
It should be safe to let this be a `lateinit` so we can loose the null assertions everywhere.